packages/ui/src/AddressView/index.tsx

```diff
@@ -4,46 +4,79 @@ import { bech32mAddress } from '@penumbra-zone/bech32m/penumbra';
 import { CopyToClipboardButton } from '../CopyToClipboardButton';
 import { AddressIcon } from './AddressIcon';
 import { Text } from '../Text';
+import { Density, useDensity } from '../utils/density';
+import cn from 'clsx';
 
 export interface AddressViewProps {
   addressView: AddressView | undefined;
   copyable?: boolean;
   hideIcon?: boolean;
+  truncate?: boolean;
 }
 
+export const getIconSize = (density: Density): number => {
+  if (density === 'compact') {
+    return 16;
+  }
+  if (density === 'slim') {
+    return 12;
+  }
+  return 24;
+};
+
+export const getFont = (density: Density): string => {
+  if (density === 'compact') {
+    return cn('flex items-center gap-1.5 text-sm text-text-primary');
+  }
+  if (density === 'slim') {
+    return cn('flex items-center gap-1 text-xs text-text-primary');
+  }
+  return cn('flex items-center gap-2 text-base text-text-primary');
+};
+
 // Renders an address or an address view.
 // If the view is given and is "visible", the account information will be displayed instead.
 export const AddressViewComponent = ({
   addressView,
   copyable = true,
   hideIcon,
+  truncate = false,
 }: AddressViewProps) => {
+  const density = useDensity();
+
   if (!addressView?.addressView.value?.address) {
     return null;
   }
 
   const addressIndex = getAddressIndex.optional(addressView);
 
-  // a randomized index has nonzero randomizer bytes
+  // A randomized index has nonzero randomizer bytes
   const isRandomized = addressIndex?.randomizer.some(v => v);
 
   const encodedAddress = bech32mAddress(addressView.addressView.value.address);
 
+  // Sub-account selector logic
+  const getAccountLabel = (index: number) =>
+    index === 0 ? 'Main Account' : `Sub-Account ${index}`;
+
   return (
-    <div className='flex items-center gap-2 text-text-primary'>
+    <div className={cn(getFont(density))}>
       {!hideIcon && (
         <div className='shrink'>
-          <AddressIcon address={addressView.addressView.value.address} size={24} />
+          <AddressIcon
+            address={addressView.addressView.value.address}
+            size={getIconSize(density)}
+          />
         </div>
       )}
 
       {addressIndex ? (
-        <Text strong truncate>
+        <Text strong truncate={truncate}>
           {isRandomized && 'IBC Deposit Address for '}
-          {`Sub-Account #${addressIndex.account}`}
+          {getAccountLabel(addressIndex.account)}
         </Text>
       ) : (
-        <Text technical truncate>
+        <Text technical truncate={truncate}>
           {encodedAddress}
         </Text>
       )}
```

Comment:
the copy button below has some stylistic inconsistencies. It displays different styles for sparse and compact densities, and it is also getting cut off. Should this be addressed at the button component level, or by adjusting the wrapping div tag to reduce its size?

---------------------

<img width="272" alt="Screenshot 2024-12-10 at 10 47 01 AM" src="https://github.com/user-attachments/assets/75b4f3f4-9b50-4175-bca4-06786a02db52">

<img width="270" alt="Screenshot 2024-12-10 at 10 47 08 AM" src="https://github.com/user-attachments/assets/1d8f7008-cd99-45ed-9628-a76e069d7e5c">

Reply:
it should be fine in the real app, it the Storybook that cuts the block. Could be solved by adding padding in the Storybook story around AddressView

Reply:
padding fixes it 👍